### ISSUE TITLE
Incorrect bracket matching in regex

### ISSUE THREAD

**Issue by dlsniper** [go-lang-plugin-org/go-lang-idea-plugin#2711]
- Plugin version (or commit hash): 9caa5d511a2b38510b86c6d8f66f54d73e29c098
IntelliJ IDEA 2016.2
Build #IC-162.1121.32, built on July 9, 2016
JRE: 1.8.0_101-b13 amd64
JVM: Java HotSpot(TM) 64-Bit Server VM by Oracle Corporation

In the following code, the square brackets are not correctly matched, see screenshot:

```go
package main

import (
	"fmt"
	"regexp"
)

func main() {
	emojiRE := regexp.MustCompile(`:[[:alnum:]]+:`)
	searchTerm := ":thumbsup:"
	searchTerm = emojiRE.ReplaceAllString(searchTerm, "")
	fmt.Printf("emoji line:%#v\n", searchTerm)
}
```

Playground link: https://play.golang.org/p/xpQYxchEhD

The code can be seen here: https://github.com/gopheracademy/gopher/blob/2a8094466ffa5a434e3bd204ea8c78c840f86145/gopher.go#L53

**Comment by zolotov** [comment]
please file an issue at youtrack.jetbrains.com

**Comment by dlsniper** [comment]
Oh, ok, will do, thank you.

**Comment by dlsniper** [comment]
btw, the fact that nothing validates against it should be reported there as well?

**Comment by zolotov** [comment]
depends on what you mean by validating

**Comment by dlsniper** [comment]
See this:
![snapshot119](https://cloud.githubusercontent.com/assets/607868/17436869/0f44c258-5b12-11e6-8d50-1792915b8153.png)

**Comment by zolotov** [comment]
yes, check regex intention also depends on intellij platform and cannot be substituted

**Comment by dlsniper** [comment]
Ah, sorry, I was too excited to get it tested I haven't had time to look at the code. Thanks.
